pkgs/tools/archivers/rar/default.nix

```diff
@@ -0,0 +1,44 @@
+{ lib, stdenv, fetchurl, autoPatchelfHook }:
+
+stdenv.mkDerivation rec {
+  pname = "rar";
+  version = "6.0.2";
+
+  src = fetchurl {
+    url = "https://www.rarlab.com/rar/rarlinux-x64-${version}.tar.gz";
+    sha256 = "sha256-WAvrUGCgfwI51Mo/RYSSF0OLPPrTegUCuDEsnBeR9uQ=";
+  };
+
+  manSrc = fetchurl {
+    url = "https://aur.archlinux.org/cgit/aur.git/plain/rar.1?h=rar&id=8e39a12e88d8a3b168c496c44c18d443c876dd10";
+    name = "rar.1";
+    sha256 = "sha256-93cSr9oAsi+xHUtMsUvICyHJe66vAImS2tLie7nt8Uw=";
+  };
+
+  dontBuild = true;
+
+  buildInputs = [ stdenv.cc.cc.lib ];
+
+  nativeBuildInputs = [ autoPatchelfHook ];
+
+  installPhase = ''
+    runHook preInstall
+
+    install -Dm755 {rar,unrar} -t "$out/bin"
+    install -Dm755 default.sfx -t "$out/lib"
+    install -Dm644 {acknow.txt,license.txt} -t "$out/share/doc/rar"
+    install -Dm644 rarfiles.lst -t "$out/etc"
+
+    install -Dm644 ${manSrc} "$out/share/man/man1/rar.1"
+
+    runHook postCheck
+  '';
+
+  meta = with lib; {
+    description = "Utility for RAR archives";
+    homepage = "https://www.rarlab.com/";
+    license = licenses.unfree;
```

Comment:
According to license:

```
  3. The software's trial version may be freely distributed, with exceptions
     noted below, provided the distribution package is not modified in any way.

     a. Nobody may distribute separate parts of the package, with the exception
        of the UnRAR components, without written permission.

     b. The software's unlicensed trial version may not be distributed
        inside of any other software package without written permission.
        The software must remain in the original unmodified installation
        file for download without any barrier and conditions to the user
        such as collecting fees for the download or making the download
        conditional on the user giving his contact data.

     c. The unmodified installation file of WinRAR must be provided pure
        and unpaired. Any bundling is interdicted. In particular the use
        of any install or download software which is providing any kind
        of download bundles is prohibited unless granted by win.rar GmbH
        in written form.

     d. Hacks/cracks, keys or key generators may not be included, pointed to
        or referred to by the distributor of the trial version.

     e. In case of violation of the precedent conditions the allowance
        lapses immediately and automatically.
```

Considering that the software must be distributed without modifications, this is probably a similar case as NVIDIA drivers so I am setting the license to `unfree` (disallowing binary distribution).

Reply:
sounds about right

Reply:
Could we set meta don't distribute (or however it is called) instead?

Reply:
`dontDistribute`? I am seeing very small amount of packages that set this, and all of them seems to be modules that takes too long to build (not really related to license issues).

BTW, `licenses.unfree` seems to be what is recommended to be set: https://github.com/master/nixpkgs/blob/4559017b7336e6b6301f05eef9aeda48f2ab9b03/doc/stdenv/meta.chapter.md#liblicensesunfreeredistributable-unfree-redistributable-liblicensesunfreeredistributable-unfree-redistributable

Reply:
(Also, just to note, it seems this docs is out-of-date. NVIDIA drivers are set to `unfreeRedistributable` these days: https://github.com/nixos/nixpkgs/blob/4559017b7336e6b6301f05eef9aeda48f2ab9b03/pkgs/os-specific/linux/nvidia-x11/generic.nix